### PR TITLE
refactor(developer): move keyman-targets.ts to common/web/types 🗜

### PR DESCRIPTION
**Pull request by mcdurdin**
Also adds unit test. Prep for use in kmc-keyboard-info.

@keymanapp-test-bot skip

### CODE REVIEW

File: common/web/types/test/kmx/test-keyman-targets.ts [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+import { AllKeymanTargets, KeymanTarget, keymanTargetsFromString } from "../../src/kmx/keyman-targets.js";
+
+describe('keyman-targets', function () {
+  it('should parse a string of targets', function() {
+    assert.deepEqual(keymanTargetsFromString('any', {expandAny: false}), [KeymanTarget.any]);
+    assert.deepEqual(keymanTargetsFromString('any', {expandAny: true}), AllKeymanTargets);
+    assert.deepEqual(keymanTargetsFromString('windows linux'), [KeymanTarget.windows, KeymanTarget.linux]);
+    assert.deepEqual(keymanTargetsFromString('windows    macosx'), [KeymanTarget.windows, KeymanTarget.macosx]);
+    assert.deepEqual(keymanTargetsFromString('windows macosx any', {expandAny: true}), AllKeymanTargets);
+  });
```
Comment (jahorton) [inline common/web/types/test/kmx/test-keyman-targets.ts:12]: Might it be worth adding a `mobile` test?  Targeting `mobile` isn't quite _any_, but it does need its own "unfolding" too.

Comment (mcdurdin) [reply common/web/types/test/kmx/test-keyman-targets.ts:12]: Right now only `any` is expanded by the function. `mobile`, `tablet`, and `desktop` are not. That would be additional functionality that is not currently required.

Comment (jahorton) [reply common/web/types/test/kmx/test-keyman-targets.ts:12]: > That would be additional functionality that is not currently required.

I'm still not clear on this.  Why do we need to expand `any` but not the others?  In particular, I see that in #9336, you've set the default target to `desktop` in the keyboard-info compiler.  Does that not need expanding?

Comment (mcdurdin) [reply common/web/types/test/kmx/test-keyman-targets.ts:12]: I've added it. I don't need it for any of the work I'm doing but it's there now.

**Review by jahorton — APPROVED** [review]

**Pull request merged**